src/python/pants/backend/docker/lint.py

```diff
@@ -0,0 +1,100 @@
+from dataclasses import dataclass
+
+from pants.backend.docker.target_types import DockerImageSources
+from pants.core.goals.lint import LintRequest, LintResult, LintResults
+from pants.core.util_rules.external_tool import (
+    DownloadedExternalTool,
+    ExternalToolRequest,
+    TemplatedExternalTool,
+)
+from pants.core.util_rules.source_files import SourceFiles, SourceFilesRequest
+from pants.engine.fs import Digest, MergeDigests
+from pants.engine.platform import Platform
+from pants.engine.process import FallibleProcessResult, Process
+from pants.engine.rules import Get, MultiGet, collect_rules, rule
+from pants.engine.target import BoolField, FieldSet, Target
+from pants.engine.unions import UnionRule
+from pants.util.logging import LogLevel
+from pants.util.strutil import pluralize
+
+
+class HadolintTool(TemplatedExternalTool):
+    options_scope = "download-hadolint"
+    name = "hadolint"
+    help = "Hadolint Dockerfile linter (https://github.com/hadolint/hadolint)"
+
+    default_version = "v2.6.0"
+    default_url_template = (
+        "https://github.com/hadolint/hadolint/releases/download/{version}/hadolint-{platform}"
+    )
+    default_url_platform_mapping = {
+        "darwin": "Darwin-x86_64",
+        "linux": "Linux-x86_64",
```

Comment:
Should be `linux_x86_64`, `macos_arm64`, and `macos_x86_64`. If there isn't a macOS ARM binary, usually the x86 one works thanks to Rosetta. I can test for you if you'd like.

Reply:
Yeah, if you don't mind, that would be great, thanks :)

Reply:
It works on Apple Silicon! 
![Screen Shot 2021-07-29 at 4 39 49 PM](https://user-images.githubusercontent.com/14852634/127578857-50ca7c32-59b9-48af-9f2b-cd0bbffc2ef5.png)